### ISSUE TITLE
Gradient boosting

### ISSUE THREAD

**Issue by hongbo-miao** [hongbo-miao/hongbomiao.com#19051]
- https://github.com/dmlc/xgboost
- https://github.com/microsoft/LightGBM
